tests/test_experimental.py

```diff
@@ -1,14 +1,392 @@
+                alpha=alpha_i,
+                correlation_threshold=lambda ensemble_size: cutoff_high,
+            )
+
+            # Compute covariances
+            prior_cov = np.cov(X, rowvar=False)
+            posterior_cutoff_low_cov = np.cov(X_i_low_cutoff, rowvar=False)
+            posterior_cutoff_high_cov = np.cov(X_i_high_cutoff, rowvar=False)
+
+            # Compute determinants of covariance matrices
+            # https://en.wikipedia.org/wiki/Generalized_variance
+            # intuitively: large determintant => high covariancce
+            #  => smaller volume of multivariate normal
+            # => less information contained in multivariate normal
+            generalized_variance_prior = np.linalg.det(prior_cov)
+            generalized_variance_low = np.linalg.det(posterior_cutoff_low_cov)
+            generalized_variance_high = np.linalg.det(posterior_cutoff_high_cov)
+
+            # The covariance is positive (semi) definite, so the determinant is >= 0
+            assert generalized_variance_low >= 0, f"1 Failed with cutoff={cutoffs}"
+
+            # Assimilating information always leads to more information,
+            # which means a smaller lower determinant
+            assert generalized_variance_low <= generalized_variance_prior
+            assert generalized_variance_high <= generalized_variance_prior
+
+            # A higher threshold means we assimilate less information
+            assert (
+                generalized_variance_low <= generalized_variance_high
+            ), f"2 Failed with cutoff_low={cutoff_low} and cutoff_high={cutoff_high}"
+
+    @pytest.mark.parametrize(
+        "linear_problem",
+        range(25),
+        indirect=True,
+        ids=[f"seed-{i+1}" for i in range(25)],
+    )
+    def test_that_adaptive_localization_works_with_dying_realizations(
+        self,
+        linear_problem,
+    ):
+        """A full worked example of the adaptive localization API, with
+        parameter groups and dying realizations (if compute nodes go down).
+
+        This is mainly meant as an example, not a test. To make it into a test
+        worth running, we set the threshold for the correlation to 0 and verify
+        that it returns the same result as ESMDA."""
+
+        def zero_correlation_threshold(ensemble_size):
+            """Adaptive localization only matches ESMDA when the threshold is <=0."""
+            return 0
+
+        # Create a problem with g(x) = A @ x
+        X, g, observations, covariance, rng = linear_problem
+        num_parameters, num_ensemble = X.shape
+        num_observations = len(observations)
+        rng = np.random.default_rng()
+
+        # Split the parameters into two groups of equal size
+        num_groups = 10
+        assert num_observations % num_groups == 0, "Num groups must divide parameters"
+        group_size = num_parameters // num_groups
+        parameters_groups = list(zip(*(iter(range(num_parameters)),) * group_size))
```

Comment:
I find this a bit hard to follow.
Could we use `np.array_split` instead? Something like this:

`np.array_split(np.arange(num_parameters), num_groups)`